mtp_api/apps/transaction/tests/test_views.py

```diff
@@ -28,26 +39,57 @@ def assertTransactionsInUpload(self, response, upload_counter):
         self.assertEqual(len(uc), 1)
         self.assertEqual(uc[0], upload_counter)
 
+    def calculate_prison_transaction_counts(self, upload_counter=None):
+        if not upload_counter:
+            upload_counter = max({x.upload_counter for x in self.transactions})
+
+        return Counter([x.prison for x in self.transactions if x.upload_counter == upload_counter])
+
     def test_list_latest_upload(self):
         url = reverse('transaction-list')
+        unique_prisons = self.calculate_prison_transaction_counts()
+        for prison, count in unique_prisons.items():
+            # we have to do if Prison because some transactions don't have a prison attached
+            # and therefore no user will be able to see those...
+            # TODO: figure out who is responsible for these transactions
```

Comment:
Ultimately, those transactions will be refunded by the `BankAdmin`s; at some point in a future sprint, we need to give them a UI to download a file (in a format we don't yet know, of course!) to upload to our bank account to issue the money back to the sender. It may be that we make a product decision *not* to do that, mind you, but that remains to be seen.